src/Models/LayerModel.vala

```diff
@@ -0,0 +1,78 @@
+/*
+* Copyright (c) 2019-2020 Alecaddd (https://alecaddd.com)
+*
+* This file is part of Akira.
+*
+* Akira is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+
+* Akira is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+
+* You should have received a copy of the GNU General Public License
+* along with Akira. If not, see <https://www.gnu.org/licenses/>.
+*
+* Authored by: Giacomo Alberini <giacomoalbe@gmail.com>
+*/
+
+public class Akira.Models.LayerModel : Models.ItemModel {
+    public string name {
+        owned get {
+            if (item.name != null) {
+                return item.name;
+            }
+
+            return item.id;
+        }
+        set {
+            item.name = value;
+        }
+    }
+
+    public string icon {
+        get {
+            return item.layer_icon;
+        }
+    }
+
+    public bool? visibility {
```

Comment:
Vala warning:
`warning: Type 'bool?' can not be used for a GLib.Object property`
Are we expecting anything else other than a boolean value here?
Also, maybe it would make more sense call this `is_visible`

Reply:
We are actually expecting a bool, but the problem is that a bool cannot be used as a property. That's why I used a `bool?` which can be used as a property, despite the warning. As you can see I'm also making use of setter and getter to update the item behind the model correcly, with a normal property getter and setter. That is to avoid using directly `model.item.locked` or `model.item.visibility`.

Actually the warning (which is build only) states something that's not completely true, since I'm using those as properties even though they _aren't supposed_ to be working as properties. It's very strange. I think that changing that would require a lot of hackish code, I'd rather stay with this warning and maybe ask help on the Gtk Discourse forum on how to get through this issue (this might also trigger a Vala change somewhere). What do you think? Alternative solution?

Wrt naming, I totally agree with you, we can change them, no problem.

Reply:
I did some research and talked with the Vala maintainer.
This was actually fixed in a recent patch landed in September: https://git.launchpad.net/vala/commit/?id=4e695a4e374df741859651f62e9a554f55991aac

So, future versions of Vala will remove this warning.
Good stuff!